core/creators/themedui.js

```diff
@@ -283,20 +283,28 @@ CKEDITOR.replaceClass = 'ckeditor';
 			outer = container;
 		}
 
+		if ( typeof width == 'string' ) {
+			width = CKEDITOR.tools.convertToPx( width );
+		}
+
 		// Set as border box width. (https://dev.ckeditor.com/ticket/5353)
 		outer.setSize( 'width', width, true );
 
 		// WebKit needs to refresh the iframe size to avoid rendering issues. (1/2) (https://dev.ckeditor.com/ticket/8348)
 		contentsFrame && ( contentsFrame.style.width = '1%' );
 
+		if ( typeof height == 'string' ) {
+			height = CKEDITOR.tools.convertToPx( height );
+		}
```

Comment:
Due to the issue where `convertToPx` works only with CSS lengths, we could convert every unit into CSS length at the beginning instead. I can suggest extracting separate function (for converting units) for readability.